### PR TITLE
KAADO-45: Implement import card page

### PR DESCRIPTION
**Pull request by nasudon**
- Implement import card page
- Implement my sharing page

### CODE REVIEW

File: src/app/sharing/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const handleAccept = async () => {
+        'use server';
+        return await executeSharing(params.id);
+    };
+
+    return (
+        <div className="w-96 min-h-screen p-4 grid grid-rows-[auto_1fr] gap-10">
+            <TopNav left={backBtn} />
+            <div className="w-full justify-self-center flex flex-col items-center gap-5">
+                <div className="w-10/12 flex flex-col items-center gap-1">
```
Comment (tsoibet) [inline src/app/sharing/[id]/page.tsx]: This div on line 33 is not needed. 
The div under TopNav (line32) can contain 3 blocks: Title, Card image and Button. `w-full` (line32) should be changed to `w-10/12` while the `w-10/12` for Button (line48) should be changed to `w-full`. Also, you will not need the `mt-4`.

File: src/services/shareService.ts [hunk 2]
```diff
@@ -24,10 +36,13 @@ export async function getSharings() {
...
 
-        const sharings = await SharingModel.find<ISharing>({ user });
+        const sharings = await SharingModel.find<SharingInfo>({ user })
+            .populate('user', 'name')
+            .populate('recipient', 'name')
+            .populate('card', 'image_front');
 
         return {
-            sharings,
+            sharings: JSON.parse(JSON.stringify(sharings)) as SharingInfo[],
```
Comment (tsoibet) [inline src/services/shareService.ts:45]: Is it for getting rid of the warning about plain objects?

**Review by tsoibet — CHANGES_REQUESTED** [review]
Other than the comments on codes, I have two other concerns:
1) Do we need to handle when a user accepts his own card?
2) Can a user share the same card to two different users at the same time? (because sharing link is created by card id)

File: src/app/sharing/[id]/page.tsx (hunk 1 above)
Comment (nasudon) [reply src/app/sharing/[id]/page.tsx]: Fixed: https://github.com/tsoibet/kaado/pull/25/commits/eac871a681175792a084b5172c847a387f29b69f

File: src/services/shareService.ts (hunk 2 above)
Comment (nasudon) [reply src/services/shareService.ts:45]: No. There was an stack overflow error when Nextjs try to parse the object to JSON to return to the frontend. This is a work around to deal with that.

**Comment by nasudon** [comment]
> Other than the comments on codes, I have two other concerns:
> 
> 1. Do we need to handle when a user accepts his own card?
> 2. Can a user share the same card to two different users at the same time? (because sharing link is created by card id)

1. We can create a ticket to handle that.
2. Share link is created by the id of "share" documents so it should be fine to share a same card with multiple people. https://github.com/tsoibet/kaado/blob/main/src/services/shareService.ts#L59-L62

**Review by tsoibet — APPROVED** [review]
I see. Sorry for getting it wrong, and thank you for answering.
LGTM!

**Pull request merged**